pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -184,18 +188,33 @@ func (e *EndpointController) Run(workers int, stopCh <-chan struct{}) {
 
 func (e *EndpointController) getPodServiceMemberships(pod *v1.Pod) (sets.String, error) {
 	set := sets.String{}
-	services, err := e.serviceLister.GetPodServices(pod)
+	allServices, err := e.serviceLister.Services(pod.Namespace).List(labels.Everything())
 	if err != nil {
 		// don't log this error because this function makes pointless
 		// errors when no services match.
 		return set, nil
 	}
-	for i := range services {
-		key, err := controller.KeyFunc(services[i])
-		if err != nil {
-			return nil, err
+
+	for i := range allServices {
+		service := allServices[i]
```

Comment:
We don't need this one, because allServices is a list of pointers (so we don't do expensive copy).
It should be enough to do:
```
for _, service := range allServices {
```

Reply:
Good catch, done 👍